### DISCUSSION TITLE
HomeKit Keypad

### DISCUSSION THREAD

**Discussion by ghost** [MiguelRipoll23/homebridge-securitysystem#92] [General]
HomeKit Keypad
Anyone aware of any keypad that can go by front door to securely set Homebridge security system to away or home. 

Enter code, set to away with delay to allow time to leave home
Enter code, set to home before delay set for alarm triggering runs out

any thoughts would be good.

**Comment by ashpole** [comment]
I use an NFC tag at the front door, I tap on and tap off.  If I forget to tap on, then a geofence trigger sets the alarm to away - just in case. 

📤 Sent from my iPad Pro

**Reply by ghost** [reply]
How do you have it set up. Currently If I swipe tag in hallway to set alarm to “away”. If I then open front door to leave , the alarm would trigger. 

What’s best setup to run delays

**Reply by MiguelRipoll23** [reply]
As @ashpole suggests, an NFC is the recommended way to arm/disarm the security system. In order to not trigger the alarm when you about to leave, on the plugin settings set `Trigger Seconds` to anything above 30 seconds.

**Comment by ashpole** [comment]
There is a setting called “arm_seconds” that I’ve got set to 30 seconds, it gives me 30 secs to exit before the system finally goes into Away mode. 



📤 Sent from my iPad Pro

**Comment by ashpole** [comment]
I should have added that you can set “trigger_seconds” to delay the alarm sounding when you come back into your home.

**Comment by ghost** [comment]
Thanks all. I have now set alarm to arm and disarm if both myself and wife leave the home. This only works if we both take phones. I like the idea of a keypad panel by door but I’m thinking there is no such thing that would interstate into HomeKit. Never mind. Thanks again

**Reply by tdallas-ger** [reply]
I just startet setting up my Security System, but already solved  the Keypad problem. 
![E0E857D6-D2D6-4842-8B28-9D7D36763B54](https://user-images.githubusercontent.com/82779683/115159828-51ed2780-a095-11eb-8edf-c948f54b21af.jpeg)
I can arm / disarm the System via code or NFC Tag at my Keycain.
